libsolidity/analysis/ControlFlowBuilder.cpp

```diff
@@ -321,8 +324,20 @@ bool ControlFlowBuilder::visit(ModifierInvocation const& _modifierInvocation)
 	auto modifierDefinition = dynamic_cast<ModifierDefinition const*>(
 		_modifierInvocation.name().annotation().referencedDeclaration
 	);
-	if (!modifierDefinition) return false;
-	if (!modifierDefinition->isImplemented()) return false;
+
+	if (!modifierDefinition)
+		return false;
```

Comment:
Is this really still right? I would guess that there may be no modifier definition at the invocation, but after virtual lookup you'll still get one...

```
abstract contract C {
        modifier mod() virtual;
        function f() mod() public {
        }
}

contract D is C {
        modifier mod() override {
                revert();
                _;
        }
}
```
My guess would be that this would report "unreachable code" for ``f`` relative to ``D``, but only if you remove this check here (respectively move it down to after the virtual lookup).

Reply:
Ah, no, only the definition allows for the virtual lookup anyways apparently, so nevermind...